File: flutter_inappwebview_forge_android/android/src/main/kotlin/com/emirkanacar/flutter_inappwebview_forge_android/download_job/DownloadJobManager.kt
package com.emirkanacar.flutter_inappwebview_forge_android.download_job

import com.emirkanacar.flutter_inappwebview_forge_android.InAppWebViewFlutterPlugin
import com.emirkanacar.flutter_inappwebview_forge_android.types.Disposable
import java.util.HashMap

open class DownloadJobManager(plugin: InAppWebViewFlutterPlugin) : Disposable {
    @JvmField
    var plugin: InAppWebViewFlutterPlugin? = plugin

    @JvmField
    val jobs: MutableMap<String, DownloadJobController> = HashMap()

    override fun dispose() {
        val ownedJobs = ArrayList(jobs.values)
        jobs.clear()
        ownedJobs.forEach { it.dispose() }
        plugin = null
    }

    companion object {
        @JvmField
        protected val LOG_TAG = "DownloadJobManager"
    }
}

File: flutter_inappwebview_forge_macos/macos/flutter_inappwebview_forge_macos/Sources/flutter_inappwebview_forge_macos/DownloadJob/DownloadJobManager.swift
import Foundation
import FlutterMacOS
import WebKit

@available(macOS 11.3, *)
public class DownloadJobManager: NSObject, Disposable {
    var plugin: InAppWebViewFlutterPlugin?
    var jobs: [String: DownloadJobController] = [:]

    public init(plugin: InAppWebViewFlutterPlugin?) {
        super.init()
        self.plugin = plugin
    }

    public func dispose() {
        let jobValues = Array(jobs.values)
        jobs.removeAll()
        jobValues.forEach { $0.dispose() }
        plugin = nil
    }

    deinit {
        dispose()
    }
}
